api/rs/slint/mcu.md

```diff
@@ -0,0 +1,405 @@
+#  }
+# }
+# mod hal { pub struct Display; impl Display { pub fn new()-> Self {todo!()} } }
+# impl DrawTarget for hal::Display{ type Color = embedded_graphics_core::pixelcolor::Rgb565; }
+
+struct DisplayWrapper<'a, T>{
+    display: &'a mut T,
+    line_buffer: &'a mut [slint::platform::software_renderer::Rgb565Pixel],
+};
+impl<T: DrawTarget<Color = embedded_graphics_core::pixelcolor::Rgb565>>
+    slint::platform::software_renderer::LineBufferProvider for DisplayWrapper<'_, T>
+{
+    type TargetPixel = slint::platform::software_renderer::Rgb565Pixel;
+    fn process_line(
+        &mut self,
+        line: usize,
+        range: core::ops::Range<usize>,
+        render_fn: impl FnOnce(&mut [Self::TargetPixel]),
+    ) {
+        // Render into the line
+        render_fn(&mut self.line_buffer[range.clone()]);
+
+        // Send the line to the screen using DrawTarget::fill_contiguous
+        self.display.fill_contiguous(
+            &Rectangle::new(Point::new(range.start as _, line as _), Size::new(range.len() as _, 1)),
+            self.line_buffer[range.clone()].iter().map(|p| RawU16::new(p.0).into())
+        ).map_err(drop).unwrap();
+    }
+}
+
+// Note that we use `1` as the const generic parameter for MinimalSoftwareWindow to indicate
+// the maximum age of the buffer we provide to `render_fn` inside `process_line`.
+// What's shown on the screen buffer is not in our RAM, but actually within the display itself.
+// We just need to re-render what changed since the last frame.
+let window = slint::platform::software_renderer::MinimalSoftwareWindow::<1>::new();
+
+const DISPLAY_WIDTH: usize = 320;
+let mut line_buffer = [slint::platform::software_renderer::Rgb565Pixel(0); DISPLAY_WIDTH];
+
+let mut display = hal::Display::new(/*...*/);
+
+// ... rest of initialization ...
+
+loop {
+    // ...
+    window.draw_if_needed(|renderer| {
+        renderer.render_by_line(DisplayWrapper{
+            display: &mut display,
+            line_buffer: &mut line_buffer
+        });
+    });
+    // ...
+}
+
+```
+
+Hint: In our experience, using then synchronous `DrawTarget::fill_contiguous` function is slow. If
+your device is capable of using DMA, you may be able to achieve better performance by using
+two line buffers: One buffer to render into with the CPU, while the other buffer is transferred to
+the screen using DMA asynchronously.
+
+Hint: If you see wrong colors on your device, it may be necessary to invert the bits in the u16 pixel
+representation before sending them to the screen driver.
+
+## Example Implementations
+
+Slint's own examples use a helper crate called `mcu-board-support` that provides implementations of
+the `Platform` trait for some MCUs, along with support for touch input and system timers.
+
+You can find the crate in our Git repository at:
+
+    <https://github.com/slint-ui/slint/tree/master/examples/mcu-board-support>
```

Comment:
```suggestion
<https://github.com/slint-ui/slint/tree/master/examples/mcu-board-support>
```

Remember that in markdown, a 4 space indentation is like a block of code.

Reply:
Thanks, that explains the build error (odd that I didn't see it locally).